File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-typescript/composed/src/providers-module.ts
import type { JsonObject, QueryParams } from './types.js';
import type { CrawChatAdminSdkContext } from './sdk-context.js';

export class CrawChatAdminProvidersModule {
  constructor(private readonly context: CrawChatAdminSdkContext) {}

  getBindings(params?: QueryParams): Promise<JsonObject> {
    return this.context.backendClient.providers.getProviderBindings(params);
  }

  upsertBindingPolicy(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.providers.upsertProviderBindingPolicy(body);
  }

  getPolicyHistory(): Promise<JsonObject> {
    return this.context.backendClient.providers.getProviderPolicyHistory();
  }

  getPolicyDiff(params: QueryParams): Promise<JsonObject> {
    return this.context.backendClient.providers.getProviderPolicyDiff(params);
  }

  previewPolicy(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.providers.previewProviderPolicy(body);
  }

  rollbackPolicy(body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.providers.rollbackProviderPolicy(body);
  }

  getRegistry(): Promise<JsonObject> {
    return this.context.backendClient.providers.getProviderRegistry();
  }
}

File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-flutter/composed/lib/src/providers_module.dart
import 'package:craw_chat_admin_backend_sdk/craw_chat_admin_backend_sdk.dart';

import 'context.dart';
import 'types.dart';

class CrawChatAdminProvidersModule {
  final CrawChatAdminSdkContext context;

  CrawChatAdminProvidersModule(this.context);

  Future<JsonObject> getBindings([CrawChatAdminQueryParams? params]) {
    return context.backendClient.providers.getProviderBindings(params);
  }

  Future<JsonObject> upsertBindingPolicy(JsonObject body) {
    return context.backendClient.providers.upsertProviderBindingPolicy(body);
  }

  Future<JsonObject> getPolicyHistory() {
    return context.backendClient.providers.getProviderPolicyHistory();
  }

  Future<JsonObject> getPolicyDiff(CrawChatAdminQueryParams params) {
    return context.backendClient.providers.getProviderPolicyDiff(params);
  }

  Future<JsonObject> previewPolicy(JsonObject body) {
    return context.backendClient.providers.previewProviderPolicy(body);
  }

  Future<JsonObject> rollbackPolicy(JsonObject body) {
    return context.backendClient.providers.rollbackProviderPolicy(body);
  }

  Future<JsonObject> getRegistry() {
    return context.backendClient.providers.getProviderRegistry();
  }
}
